### PR TITLE
Notifications on add/remove device

### PR DESCRIPTION
**Pull request by nalinbhardwaj**
not tested
based on #384 will merge after

### CODE REVIEW

File: packages/daimo-common/test/keySlot.test.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import assert from "node:assert";
+import test from "node:test";
+
+import { SlotType, findUnusedSlot, getSlotLabel } from "../src/keySlot";
+
+const testCases: [number, string][] = [
+  [0, "Mobile A"],
+  [25, "Mobile Z"],
+  [26, "Mobile AA"],
+  [129, "Backup B"],
```
Comment (dcposch) [inline packages/daimo-common/test/keySlot.test.ts:10]: maybe call these `Passkey A` / `Passkey B` ? apple native UI will surface them as passkeys whenever they're used

File: packages/daimo-common/test/keySlot.test.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import assert from "node:assert";
+import test from "node:test";
+
+import { SlotType, findUnusedSlot, getSlotLabel } from "../src/keySlot";
+
+const testCases: [number, string][] = [
+  [0, "Mobile A"],
+  [25, "Mobile Z"],
+  [26, "Mobile AA"],
```
Comment (dcposch) [inline packages/daimo-common/test/keySlot.test.ts:9]: add `Computer A`

File: packages/daimo-api/src/contract/keyRegistry.ts [hunk 3]
```diff
@@ -96,6 +114,10 @@ export class KeyRegistry {
           addedAt: Number(log.blockNumber),
           slot,
         });
+
+        if (this.addrToDeploymentTxHash.get(addr) === undefined) {
```
Comment (dcposch) [inline packages/daimo-api/src/contract/keyRegistry.ts:118]: nit: `if (!this.addrToDeploymentTxHash.has(...`

File: packages/daimo-api/test/pushNotifier.test.ts [hunk 4]
```diff
@@ -119,6 +123,76 @@ test("PushNotifier", async () => {
...
+      }),
+    ];
+    const output = pn.getPushMessagesFromKeyRotations(input);
+
+    assert.strictEqual(output.length, 1);
+    assert.deepStrictEqual(output[0].to, ["pushTokenAlice"]);
+    assert.strictEqual(output[0].title, "Mobile A removed");
+    assert.strictEqual(
+      output[0].body,
+      "You removed Mobile A from your account"
```
Comment (dcposch) [inline packages/daimo-api/test/pushNotifier.test.ts:148]: Phone A

File: packages/daimo-common/test/keySlot.test.ts (hunk 2 above)
Comment (dcposch) [reply packages/daimo-common/test/keySlot.test.ts:9]: so we'd have `Phone A` / `Computer A` / `Passkey A` or `Passkey Backup A` 

lmk if that looks right

**Review by dcposch — APPROVED** [review]
LGTM after comments

File: packages/daimo-common/test/keySlot.test.ts (hunk 2 above)
Comment (nalinbhardwaj) [reply packages/daimo-common/test/keySlot.test.ts]: done in #383

File: packages/daimo-api/src/contract/keyRegistry.ts (hunk 3 above)
Comment (nalinbhardwaj) [reply packages/daimo-api/src/contract/keyRegistry.ts]: done

File: packages/daimo-api/test/pushNotifier.test.ts (hunk 4 above)
Comment (nalinbhardwaj) [reply packages/daimo-api/test/pushNotifier.test.ts]: done in #383

**Pull request merged**